### PR TITLE
25 positive delta v

### LINKED ISSUE #25: positive delta v
safe_autonomy_sims.pettingzoo.inspection.utils.py delta_v function has `np.abs(v_norm - prev_v_norm)` instead of v_norm - prev_v_norm, resulting in positive values for the delta v penalty

### PR DESCRIPTION
**Pull request by JohnMcCarroll**
In this PR:
- abs difference in delta_v for pettingzoo + gymnasium envs
- unit tests to ensure delta_v computed correctly

### CODE REVIEW

**Comment by JohnMcCarroll** [comment]
Two issues keeping this from merge:
- Should delta_v be left alone, but absolute value calculated just for rewards?
- new delta_v unit tests need to be integrated with the rest of the test suite.

File: test/unit_tests/safe_autonomy_sims/conftest.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+# test case parsing (TODO: import from saferl.conftest)
+delimiter = ","
+
+def execute_strings(value):
+    """
+    A recursive helper function to convert string expressions (read from a file) into numerical floats.
+
+    Parameters
+    ----------
+    value
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/conftest.py]: Add input type(s) to docstring

File: test/unit_tests/safe_autonomy_sims/conftest.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        A value to be converted or modified based on string format.
+
+    Returns
+    -------
+    value
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/conftest.py]: Add return type(s) to docstring

File: test/unit_tests/safe_autonomy_sims/conftest.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import pytest
+import numpy as np
+import yaml
+
+
+# test case parsing (TODO: import from saferl.conftest)
+delimiter = ","
+
+def execute_strings(value):
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/conftest.py]: Add type hints to input

File: test/unit_tests/safe_autonomy_sims/conftest.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            pass
+    elif type(value) is dict:
+        for k, v in value.items():
+            value[k] = execute_strings(v)
+    elif type(value) is list:
+        for i in range(0, len(value)):
+            value[i] = execute_strings(value[i])
+    return value
+
+def preprocess_values(value, keyword, functions):
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/conftest.py]: Add type hints to input

File: test/unit_tests/safe_autonomy_sims/conftest.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    """
+    A recursive helper function to
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/conftest.py]: Complete function description

File: test/unit_tests/safe_autonomy_sims/conftest.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+    Parameters
+    ----------
+    value
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/conftest.py]: Add input type(s) to docstring

File: test/unit_tests/safe_autonomy_sims/conftest.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        A value to be converted or modified based on string format.
+
+    Returns
+    -------
+    value
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/conftest.py]: Add return type(s) to docstring

File: test/unit_tests/safe_autonomy_sims/conftest.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        function_string = functions[keyword].split()[0]  # clean input
+        function_string += "({})".format(value)
+        try:
+            value = eval(function_string)
+        except Exception as e:
+            print(e)
+
+    return value
+
+def read_test_cases(file_path, parameter_keywords):
```
Comment (jamie-cunningham) [inline test/unit_tests/safe_autonomy_sims/conftest.py]: Add type hints to input

**Review by jamie-cunningham — CHANGES_REQUESTED** [review]
Overall looks great. Should be fine to merge once review comments are addressed.

**Review by jamie-cunningham — APPROVED** [review]

**Pull request merged**